game.go

```diff
@@ -205,25 +216,34 @@ func (g *Game) currentTeam() Team {
 }
 
 func newGame(id string, state GameState) *Game {
-	rnd := rand.New(rand.NewSource(state.Seed))
+	// Reset seed and index if game has exhausted all words
+	if (state.PermIndex + wordsPerGame >= len(state.WordSet)) {
+		state = randomState(state.WordSet)
+	}
+
+	// used for generating words, consistent randomness across games
+	seedRnd := rand.New(rand.NewSource(state.Seed))
+	// used for all other random settings. starting team and shuffling
+	randRnd := rand.New(rand.NewSource(rand.Int63()))
```

Comment:
Previously `newGame` was a pure function. Every time it was called with the same input, it produced an identical `*Game`. It'd be nice to preserve that contract. Since this line introduces randomness _not_ deterministically derived from `state.Seed`, this breaks that invariant.

Reply:
fixed